src/Catel.Analyzers/Supressors/IDE0051OnPropertyChangeSupressor.cs

```diff
@@ -0,0 +1,159 @@
+
+                    if (!accessModifierNode.IsKind(Microsoft.CodeAnalysis.CSharp.SyntaxKind.PrivateKeyword))
+                    {
+                        // Skip non-private
+                        continue;
+                    }
+
+                    if (targetSymbol is IMethodSymbol method && method.Name.StartsWith("On") && method.Name.EndsWith("Changed"))
+                    {
+                        var propertyName = method.Name.ReplaceFirst("On", "").ReplaceLast("Changed", "");
+
+                        // Search for property
+                        // Step 1: Search by property name through class properties
+                        // Step 2: Try to check Expose attributes:
+                        //  - arguments passed to ctor
+                        //  - named arguments
+                        // This call is more expensive than queries on syntax tree
+
+                        if (containerClassSymbol.TryFindProperty(propertyName, out var weavedPropertySymbol))
+                        {
+                            context.ReportSuppression(Suppression.Create(supresssionDescriptor, diagnostic));
+                            return;
+                        }
+
+                        if (fodyExposeType is null)
+                        {
+                            continue;
+                        }
+
+                        var exposedMarkedDeclarations = from descendantNode in containerClassSyntax.DescendantNodes(x => x is ClassDeclarationSyntax || x is PropertyDeclarationSyntax || x is AttributeListSyntax)
+                                                        where descendantNode is AttributeSyntax && string.Equals(GetIdentifier(descendantNode as AttributeSyntax), FodyAttributeLookupName)
+                                                        select descendantNode.FirstAncestor<PropertyDeclarationSyntax>();
+
+                        foreach (var property in exposedMarkedDeclarations.Distinct())
+                        {
+                            var propertySymbol = rootSemantic.GetDeclaredSymbol(property, default);
+
+                            var exposeAttributesList = propertySymbol.GetAttributes().Where(x => SymbolEqualityComparer.Default.Equals(x.AttributeClass, fodyExposeType)).ToList();
+
+                            foreach (var exposeAttribute in exposeAttributesList)
+                            {
+                                if (exposeAttribute.ConstructorArguments.Any())
+                                {
+                                    var constructorArgValue = exposeAttribute.ConstructorArguments.FirstOrDefault().Value.ToString();
+
+                                    if (string.Equals(constructorArgValue, propertyName))
+                                    {
+                                        context.ReportSuppression(Suppression.Create(supresssionDescriptor, diagnostic));
+                                        return;
```

Comment:
I don't think return is correct here. What about the other diagnostics we need to go through?

We could extract this separate for-loop (for fody exposed checking) into a separate method and either return true or false and continue based on the return value.

Reply:
Right, should be continue into nesting loop here. Need to rewrite this part